Source/Dafny/Translator.cs

```diff
@@ -18319,6 +18319,9 @@ private class TriggersExplorer : BottomUpVisitor {
       if (expr is FunctionCallExpr) {
         var e = (FunctionCallExpr)expr;
         Util.Concat(e.TypeApplication_AtEnclosingClass, e.TypeApplication_JustFunction).Iter(ty => ComputeFreeTypeVariables(ty, fvs));
+      } else if (expr is LetExpr) {
+        var e = (LetExpr)expr;
```

Comment:
Shall we use the pattern matching syntax?

```
if (expr is LetExpr e) {
 e.BoundVars.IteR(..);
}
```

I'm curious, doesn't your IDE suggest this?

Also, I prefer more verbose variable names like `letExpr` instead of `e`, but it's not a strong opinion.

Reply:
Yes, I tend to use this when there is just one such `if` branch. But this feature is broken in C#, in that the scope of the variable `e` you introduce is scoped outside the `if`. Therefore, if you have several branches, you have to make up different names for the `e` in each one, which I don't like. That is, I find it more of a mental burden to have to come up with a new name and to remember which of the names I made up that I'm working with than to use the old `var e =` syntax on each line.